### PR TITLE
Add babel and distribute compiled (ES5) version of module as default

### LINKED ISSUE #3: Module parse failed: Unexpected token (5:22)
I'm using this with CRNA and there is a build error on `npm run web`:

```log
ERROR in ./node_modules/react-native-web-linear-gradient/src/index.js
Module parse failed: Unexpected token (5:22)
You may need an appropriate loader to handle this file type.
|
| export default class LinearGradient extends PureComponent {
|   static defaultProps = {
|     start: {
|       x: 0.5,
 @ ./src/Main.js 1:1366-1405
 @ ./src/App.js
 @ ./src/index.js
 @ multi (webpack)-dev-server/client?http://localhost:8080 webpack/hot/dev-server ./src/index.js
webpack: Failed to compile.
```

### PR DESCRIPTION
**Pull request by cymen**
Resolves #3 and #1. Comment from #3:

> Typically on web projects, anything inside of node_modules is not processed by babel -- just things within the project. Typically, published modules are compiled once.

So this should save people some time.

### CODE REVIEW

**Comment by cymen** [comment]
@AuroreM @Minishlink Any chance of getting this in? I had to fork with these changes and publish it internally on a project but I'd like to switch back to this.

File: package.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
   "name": "react-native-web-linear-gradient",
-  "version": "1.0.2",
+  "version": "1.0.3",
```
Comment (Minishlink) [inline package.json]: Can you keep the 1.0.2
It will be bumped when I'll publish

File: dist/index.js [hunk 2]
```diff
@@ -0,0 +1 @@
...ient)).call.apply(_ref,[this].concat(args))),_this),_this.state={width:1,height:1},_this.measure=function(_ref2){var nativeEvent=_ref2.nativeEvent;return _this.setState({width:nativeEvent.layout.width,height:nativeEvent.layout.height});},_this.getAngle=function(){var angle=Math.atan2(_this.state.width*(_this.props.end.y-_this.props.start.y),_this.state.height*(_this.props.end.x-_this.props.start.x))+Math.PI/2;return angle+'rad';},_this.getColors=function(){return _this.props.colors.map(function(color,index){var location=_this.props.locations[index];var locationStyle='';if(location){locationStyle=' '+location*100+'%';}return color+locationStyle;}).join(',');},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(LinearGradient,[{key:'render',value:function render(){return _react2.default.createElement(_reactNative.View,{style:[this.props.style,{backgroundImage:'linear-gradient('+this.getAngle()+','+this.getColors()+')'}],onLayout:this.measure,__source:{fileName:_jsxFileName,lineNumber:54}},this.props.children);}}]);return LinearGradient;}(_react.PureComponent);LinearGradient.defaultProps={start:{x:0.5,y:0},end:{x:0.5,y:1},locations:[],colors:[]};exports.default=LinearGradient;
```
Comment (Minishlink) [inline dist/index.js]: Could you revert this change please - not commit `dist` folder
This is bad practice because it makes git diffs hard to read and beginners might contribute to dist files unknowingly

**Comment by Minishlink** [comment]
By the way, I don't recommend using/publishing ES5 versions:

- you'll use some older syntax
  - so this won't be optimized for new browsers
  - even when you might not care for older browsers
  - the build size will be bigger
- this slows down adoption of newer standards

[Apparently](https://babeljs.io/blog/2018/06/26/on-consuming-and-publishing-es2015+-packages) Babel authors recommend to also publish last syntax version on main-es. Can you do that?

File: dist/index.js (hunk 2 above)
Comment (cymen) [reply dist/index.js]: This is the crux of the issue -- the reason I opened this PR. I cannot exclude dist because look at package.json:

```
"main": "dist/index.js"
```

If I exclude it, `main` will point to a file that doesn't exist.

**Comment by cymen** [comment]
@Minishlink I can make those changes however I `dist` needs to be included. Per your link:

```
We should continue to publish ES5/CJS under main for backwards compatibility with current tooling but also publish a version compiled down to latest syntax (no experimental proposals) under a new key we can standardize on like main-es. (I don't believe module should be that key since it was intended only for JS Modules).
```

Do you agree? If so, I can update. If not, I'll close this PR.

Thanks!

**Comment by cymen** [comment]
I've updated per the PR discussion above leaving `dist` so `main` can point to `dist/index.js`, removing the version bump and adding `main-es` (which points to `src/index.js`).

**Comment by iamhosseindhv** [comment]
I believe this PR is of high importance. Not sure why this has not been solved already or even now that we have this PR, we're still waiting for the fix to be published!

~500 downloads of this package every week. so much time being wasted on this issue.

**Comment by Minishlink** [comment]
@iamhosseindhv No need for that tone. Besides that, if your setup need these changes right now, there are several ways including patching or forking.
